airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry_report.py

```diff
@@ -22,148 +105,185 @@ def augment_and_normalize_connector_dataframes(
     # Add a column 'is_oss' to indicate if an image/version pair is in the oss registry
     oss_df["is_oss"] = True
 
-    composite_key = [primaryKey, "dockerRepository", "dockerImageTag"]
+    # Merge the two registries on the 'image' and 'version' columns
+    total_registry = pd.merge(oss_df, cloud_df, how="outer", suffixes=(OSS_SUFFIX, CLOUD_SUFFIX), on=primaryKey)
 
-    # Merge the two registries on the 'image' and 'version' columns, keeping only the unique pairs
-    total_registry = pd.merge(cloud_df, oss_df, how="outer", on=composite_key).drop_duplicates(subset=composite_key)
+    # remove duplicates from the merged dataframe
+    total_registry = total_registry.drop_duplicates(subset=primaryKey, keep="first")
 
     # Replace NaN values in the 'is_cloud' and 'is_oss' columns with False
     total_registry[["is_cloud", "is_oss"]] = total_registry[["is_cloud", "is_oss"]].fillna(False)
 
-    registry_with_metadata = pd.merge(
-        total_registry,
-        valid_metadata_report_dataframe[["definitionId", "is_metadata_valid"]],
-        left_on=primaryKey,
-        right_on="definitionId",
-        how="left",
-    )
-
-    # merge with cached_specs on dockerRepository and dockerImageTag
-    cached_specs["is_spec_cached"] = True
-    merged_registry = pd.merge(
-        registry_with_metadata,
-        cached_specs,
-        left_on=["dockerRepository", "dockerImageTag"],
-        right_on=["docker_repository", "docker_image_tag"],
-        how="left",
-    )
-
     # Set connectorType to 'source' or 'destination'
-    merged_registry["connector_type"] = connector_type
-    merged_registry["is_source_controlled"] = merged_registry["dockerRepository"].apply(
-        lambda x: x.lstrip("airbyte/") in github_connector_folders
-    )
+    total_registry["connector_type"] = connector_type
 
-    return merged_registry
+    total_registry["github_url"] = total_registry["dockerRepository_oss"].apply(lambda x: github_url(x, github_connector_folders))
 
+    total_registry["issue_url"] = total_registry.apply(issue_url, axis=1)
+    total_registry["test_summary_url"] = total_registry.apply(test_summary_url, axis=1)
 
-# ASSETS
-
+    # Merge docker repo and version into separate columns
+    total_registry["docker_image_oss"] = total_registry.apply(lambda x: merge_docker_repo_and_version(x, OSS_SUFFIX), axis=1)
+    total_registry["docker_image_cloud"] = total_registry.apply(lambda x: merge_docker_repo_and_version(x, CLOUD_SUFFIX), axis=1)
+    total_registry["docker_images_match"] = total_registry.apply(lambda x: x["docker_image_oss"] == x["docker_image_cloud"], axis=1)
```

Comment:
```suggestion
    total_registry["docker_images_match"] = total_registry["docker_image_oss"] == total_registry["docker_image_cloud"]
```
Apply operations on DF are expensive, it's better to avoid them when possible.